opentech/public/projects/models.py

```diff
@@ -0,0 +1,166 @@
+    RelatedPage,
+)
+
+from .widgets import CategoriesWidget
+
+
+class ProjectContactDetails(models.Model):
+    project_page = ParentalKey(
+        'ProjectPage',
+        related_name='contact_details'
+    )
+    site_titles = (
+        ('website', "Main Website URL"),
+        ('twitter', "Twitter Handle"),
+        ('github', "Github Organisation or Project"),
+    )
+    site_urls = (
+        ('website', ''),
+        ('twitter', 'https://twitter.com/'),
+        ('github', 'https://github.com/'),
+    )
+    service = models.CharField(
+        max_length=200,
+        choices=site_titles,
+    )
+    value = models.CharField(max_length=255)
+
+    @property
+    def url(self):
+        return dict(self.site_urls)[self.service] + self.value
+
+    def service_name(self):
+        site_display = {
+            'twitter': '@' + self.value,
+            'github': 'Github',
+            'website': 'Main Website',
+        }
+        return site_display[self.service]
+
+    def clean(self):
+        if self.service == 'twitter' and self.value.startswith('@'):
+            self.username = self.username[1:]
+
+        if self.service == 'website':
+            validate = URLValidator()
+            try:
+                validate(self.value)
+            except ValidationError as e:
+                raise ValidationError({'value': e})
+
+
+class ProjectPageRelatedPage(RelatedPage):
+    source_page = ParentalKey('ProjectPage', related_name='related_pages')
+
+    panels = [
+        PageChooserPanel('page', 'projects.ProjectPage'),
+    ]
+
+
+class ProjectFunding(BaseFunding):
+    page = ParentalKey('ProjectPage', related_name='funding')
+
+
+class ProjectPage(FundingMixin, BasePage):
+    STATUSES = (
+        ('idea', "Just an Idea. (Pre-alpha)"),
+        ('exists', "It Exists! (Alpha/Beta)"),
+        ('release', "It's basically done. (Release)"),
+        ('production', "People Use it. (Production)"),
+    )
+
+    subpage_types = []
+    parent_page_types = ['ProjectIndexPage']
+
+    introduction = models.TextField(blank=True)
+    icon = models.ForeignKey(
+        'images.CustomImage',
+        null=True,
+        blank=True,
+        related_name='+',
+        on_delete=models.SET_NULL
+    )
+    status = models.CharField(choices=STATUSES, max_length=25, default=STATUSES[0][0])
+    body = StreamField(StoryBlock())
+
+    categories = models.TextField(default='{}', blank=True)
+
+    search_fields = BasePage.search_fields + [
+        index.SearchField('introduction'),
+        index.SearchField('body'),
+    ]
+
+    content_panels = BasePage.content_panels + [
+        ImageChooserPanel('icon'),
+        FieldPanel('introduction'),
+        FieldPanel('status'),
+        StreamFieldPanel('body'),
+        InlinePanel('contact_details', label="Contact Details"),
+        InlinePanel('related_pages', label="Related Projects"),
+    ] + FundingMixin.content_panels + [
+        FieldPanel('categories', widget=CategoriesWidget),
```

Comment:
Make this collapsible and possibly collapsed by default?

Other than this/ FE order, :shipit:

Reply:
![image](https://user-images.githubusercontent.com/6338661/35151445-3da37bd8-fd16-11e7-9be5-bb8c48879490.png)

Reply:
Looks a little weird with the duplicate name, but not terrible.